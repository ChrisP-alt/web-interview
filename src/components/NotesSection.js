import React from 'react'
import PropTypes from 'prop-types'

import Section from './Section'
import { ReactComponent as Icon } from '../icons/comments-regular.svg'

import styles from '../styles/components/notesSection.module.scss'

const NotesSection = ({ handleNotesChange, notes }) => {
  return (
    <Section headerText="Notes" Icon={Icon}>
      <textarea
        className={styles.textArea}
        onChange={handleNotesChange}
        placeholder="Describe your symptoms"
        rows={6}
        value={notes}
      />
    </Section>
  )
}

NotesSection.propTypes = {
  handleNotesChange: PropTypes.func.isRequired,
  notes: PropTypes.string,
}

export default NotesSection
