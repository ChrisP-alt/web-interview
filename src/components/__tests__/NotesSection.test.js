import React from 'react'
import { shallow } from 'enzyme'

import NotesSection from '../NotesSection'

describe('NotesSection component', () => {
  it('renders correctly with no notes', () => {
    const wrapper = shallow(
      <NotesSection handleNotesChange={jest.fn()} notes="" />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with notes', () => {
    const notes = 'Notes on my appointment'
    const wrapper = shallow(
      <NotesSection handleNotesChange={jest.fn()} notes={notes} />
    )
    expect(wrapper.find('textarea').props().value).toEqual(notes)
  })

  it('calls handleNotesChange when the text is changed', () => {
    const handleNotesChange = jest.fn()
    const wrapper = shallow(
      <NotesSection handleNotesChange={handleNotesChange} notes="" />
    )
    const event = { target: { value: 'Updated notes' } }
    wrapper.find('textarea').simulate('change', event)
    expect(handleNotesChange).toHaveBeenCalledWith(event)
  })
})
