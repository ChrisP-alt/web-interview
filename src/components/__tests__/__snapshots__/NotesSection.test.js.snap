// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesSection component renders correctly with no notes 1`] = `
<Section
  Icon={[Function]}
  headerText="Notes"
>
  <textarea
    className="textArea"
    onChange={[MockFunction]}
    placeholder="Describe your symptoms"
    rows={6}
    value=""
  />
</Section>
`;
